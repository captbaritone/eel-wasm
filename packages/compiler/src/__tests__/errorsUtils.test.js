const fs = require("fs");
const path = require("path");
const { compileModule } = require("../compiler");
const shims = require("../shims");

const DIR = path.join(__dirname, "../../fixtures/errors");

const testCases = fs.readdirSync(DIR);

testCases.forEach(filename => {
  const testFunc = filename.startsWith("only.") ? test.only : test;
  const testName = filename.replace(/^only\./, "");
  testFunc(`${testName}`, () => {
    const eel = fs.readFileSync(path.join(DIR, filename), { encoding: "utf8" });
    const compile = () => {
      compileModule({ globals: new Set(), functions: { run: eel }, shims });
    };

    expect(compile).toThrowErrorMatchingSnapshot();
  });
});
