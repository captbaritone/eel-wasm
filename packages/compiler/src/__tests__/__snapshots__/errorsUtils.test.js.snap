// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assign_non_identifier.eel 1`] = `
"Expected the first argument of \`assign()\` to be an identifier.

> 1 | assign(10 - 2, 10);
    |        ^^"
`;

exports[`assign_to_buffer_too_many_args.eel 1`] = `
"Expected 1 argument when assinging to a buffer

> 1 | megabuf(2, 3) = 10;
    | ^^^^^^^^^^^^^"
`;

exports[`assign_to_non_buff_function_call.eel 1`] = `
"The only function calls which may be assigned to are \`gmegabuf()\` and \`megabuf()\`.

> 1 | abs(10) = 10;
    | ^^^"
`;

exports[`parse_error.eel 1`] = `
"Parse Error: Expecting 'EOF', got ')'

  2 | z = sqr(y, 10);
> 3 | x = assign);
    |     ^^^^^^
  4 | y = 100;
"
`;

exports[`undefined_function.eel 1`] = `
"\\"noexist\\" is not defined.

> 1 | x = noexist();
    |     ^^^^^^^"
`;

exports[`undefined_function_multiline.eel 1`] = `
"\\"noexist\\" is not defined.

  2 | z = sqr(y, 10);
> 3 | x = noexist();
    |     ^^^^^^^
  4 | y = 100;
"
`;

exports[`undefined_function_with_comments.eel 1`] = `
"\\"noexist\\" is not defined.

  3 | // Comment line three
> 4 | x = noexist();
    |     ^^^^^^^"
`;
